optuna/_optimize.py

```diff
@@ -232,6 +232,13 @@ def _run_trial(
         else:
             state = TrialState.COMPLETE
 
+    # add pruned check for catboost
+    if trial.user_attrs.get("pruned") is True:
+        step = trial.user_attrs.get("step")
+        state = TrialState.PRUNED
+        func_err = optuna.TrialPruned("Trial was pruned at epoch {}.".format(step))
+        values = None
```

Comment:
Unfortunately, this modification for catboost pruning might affect the existing code too much...